core/src/main/java/jenkins/BreadCrumbItem.java

```diff
@@ -0,0 +1,43 @@
+package jenkins;
+
+
+
+public class BreadCrumbItem {
+
+    private String displayName;
+    private boolean hasContextualMenu;
+    private BreadCrumb breadCrumb;
+
+
+    public BreadCrumbItem getFirst() {
+        return breadCrumb.generateBreadCrumbs().get(0);
+    }
+
+    public BreadCrumbItem getLast() {
+        int sizeofBreadCrumbList = breadCrumb.generateBreadCrumbs().size();
+        return breadCrumb.generateBreadCrumbs().get(sizeofBreadCrumbList - 1);
+    }
+
+    public void reNameBreadCrumb(int index, String name) {
```

Comment:
We could for now just change the method to `reNameRootBreadCrumb` if needed. Would love to get some feedback on this.